Libraries/Core/__tests__/ReactNativeVersionCheck-test.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+describe('checkVersion', () => {
+  describe('in development', () => {
+    _setDevelopmentModeForTests(true);
+    _defineCheckVersionTests();
+  });
+
+  describe('in production', () => {
+    _setDevelopmentModeForTests(false);
+    _defineCheckVersionTests();
+  });
+});
+
+function _setDevelopmentModeForTests(dev) {
+  let originalDev;
+
+  beforeAll(() => {
+    originalDev = global.__DEV__;
+    global.__DEV__ = dev;
+  });
+
+  afterAll(() => {
+    global.__DEV__ = originalDev;
+  });
+}
+
+function _defineCheckVersionTests() {
+  afterEach(() => {
+    jest.resetModules();
+  });
+
+  it('passes when all the versions are zero', () => {
+    jest.dontMock('ReactNativeVersion');
+    _mockNativeVersion(0, 0, 0);
+
+    const ReactNativeVersion = require('ReactNativeVersion');
+    const ReactNativeVersionCheck = require('ReactNativeVersionCheck');
+    expect(ReactNativeVersion).toMatchObject({
+      version: {major: 0, minor: 0, patch: 0, prerelease: null},
+    });
+    expect(() => ReactNativeVersionCheck.checkVersions()).not.toThrow();
+  });
+
+  it('passes when the minor matches when the major is zero', () => {
+    _mockJsVersion(0, 1, 0);
+    _mockNativeVersion(0, 1, 0);
+
+    const ReactNativeVersionCheck = require('ReactNativeVersionCheck');
+    expect(() => ReactNativeVersionCheck.checkVersions()).not.toThrow();
+  });
+
+  it("throws when the minor doesn't match when the major is zero", () => {
```

Comment:
Can you add a test for major don't match too so we cover all cases

Reply:
Yep, will do. We can't add a test for checking that the major matches since we don't have that logic right now, we'll need to add it later if/when we shift RN versions over by one point (RN 51, RN 52, etc...).